src/main/java/redis/clients/jedis/Observability.java

```diff
@@ -107,12 +112,14 @@ public static void annotateSpan(Span span, String description, Attribute ...attr
 
     public static class RoundtripTrackingSpan implements AutoCloseable {
         private Span span;
+        private Scope spanScope;
         private long startTimeNs; 
         private String commandName;
         private boolean closed;
 
         public RoundtripTrackingSpan(String name, String commandName) {
-            this.span = tracer.spanBuilder(name).startSpan();
+            this.spanScope = tracer.spanBuilder(name).startScopedSpan();
+            this.span = tracer.getCurrentSpan();
```

Comment:
@songy23 this is how I've done it

Reply:
You also need to close the scope after setting it, something like:
```java
spanScope.close();
span.end();
```
So it might be easier to just return the span, and set span into scope in the code that calls RoundtripTrackingSpan() instead of doing it in RoundtripTrackingSpan().